src/core/core.scale.js

```diff
@@ -333,7 +333,7 @@ module.exports = Element.extend({
 		var me = this;
 		var context = me.ctx;
 		var tickOpts = me.options.ticks;
-		var labels = labelsFromTicks(me._ticks);
+		var labels = tickOpts.autoSkip ? labelsFromTicks(me._autoSkip(me.getTicks())) : labelsFromTicks(me.getTicks());
```

Comment:
You could rewrite as:

    var labels = labelsFromTicks(tickOpts.autoSkip ? me._autoSkip(me.getTicks()) : me.getTicks());

Reply:
`_autoSkip` is pretty expensive as well. It's not entirely clear to me that this is faster than the previous behavior. I'm wondering if it can be moved to `update` so that it only happens once.

Reply:
Hmm. This is getting really gnarly: https://github.com/chartjs/Chart.js/issues/3981#issuecomment-464366411

I'm not sure what the best solution is